roles/pool/src/lib/mining_pool/mod.rs

```diff
@@ -96,6 +105,73 @@ pub struct Configuration {
     pub test_only_listen_adress_plain: String,
 }
 
+pub struct TemplateProviderConfig {
+    address: String,
+    authority_public_key: Option<Secp256k1PublicKey>,
+}
+
+impl TemplateProviderConfig {
+    pub fn new(address: String, authority_public_key: Option<Secp256k1PublicKey>) -> Self {
+        Self {
+            address,
+            authority_public_key,
+        }
+    }
+}
+
+pub struct AuthorityConfig {
+    pub public_key: Secp256k1PublicKey,
+    pub secret_key: Secp256k1SecretKey,
+}
+
+impl AuthorityConfig {
+    pub fn new(public_key: Secp256k1PublicKey, secret_key: Secp256k1SecretKey) -> Self {
+        Self {
+            public_key,
+            secret_key,
+        }
+    }
+}
+
+pub struct PoolConnection {
```

Comment:
maybe call this `ConnectionConfig`?

just to make it explicitly clear that this `struct` is just a `*Config`, not really doing anything meaningful aside from that